### ISSUE TITLE
Tuya thermostat VH Control Atlas not recognized

### ISSUE THREAD

**Issue by jaapkoppejan** [Xenomes/Domoticz-TinyTUYA-Plugin#90]
Hello,

I have trouble to retrieve the data from my VH Control Atlas thermostat into Domoticz. The sensors are there, but do not contain values. There is however communication from Domoticz to the Tuya platform, as I see the effect of pressing a switch in domoticz switch in real life. I did use DP instruction for this device.

The json dump that does contain the sensor data is attached!

Thanks a lot for your effort and best regards,

[dump.json](https://github.com/Xenomes/Domoticz-TinyTUYA-Plugin/files/14171973/dump.json)

**Comment by Xenomes** [comment]
Hi Jaap,
What is the version of the plugin?
I loaded you dump and i got this:
![afbeelding](https://github.com/Xenomes/Domoticz-TinyTUYA-Plugin/assets/11648297/36adbe80-6e2f-4216-8587-bf85ac0fbd4f)

**Comment by jaapkoppejan** [comment]
Hi Xenomes,
The version of my plugin is 1.7.3. When I select the same sensors from Domoticz I get this list:
![image](https://github.com/Xenomes/Domoticz-TinyTUYA-Plugin/assets/40127840/f35ed151-3a18-4fa2-8e4f-17f8d2aeb19b)
So the sensors are there, but there is just no update of values.

**Comment by Xenomes** [comment]
Can you do a new pull, maybe you are between two updates in the version.

**Comment by jaapkoppejan** [comment]
I tried again, however it tells me I am 'Already up to date.' so there is no difference in behaviour.

Maybe relevant to know that a power plug switch in my tuya pool of devices shows the same behaviour (one way direction of data traffic only from domoticz to tuya, not from tuya to domoticz)

 I am on Domoticz 2024.4 (stable).

**Comment by Xenomes** [comment]
I have push a new version, can you check it out?

**Comment by jaapkoppejan** [comment]
Thanks for your efforts! However, after updating and restarting Domoticz I still get no device data in Domoticz. The error log shows the following:

```
2024-02-06 21:57:53.921 Error: Tiny Tuya: Device read failed: bf56544b14bcff04e5irmc
2024-02-06 21:58:03.919 Error: Tiny Tuya: Call to function 'onCommand' failed, exception details:
2024-02-06 21:58:03.924 Error: Tiny Tuya: Traceback (most recent call last):
2024-02-06 21:58:03.924 Error: Tiny Tuya: File "/home/pi/domoticz/plugins/Domoticz-TinyTUYA-Plugin/plugin.py", line 645, in onCommand
2024-02-06 21:58:03.924 Error: Tiny Tuya: _plugin.onCommand(DeviceID, Unit, Command, Level, Color)
2024-02-06 21:58:03.924 Error: Tiny Tuya: File "/home/pi/domoticz/plugins/Domoticz-TinyTUYA-Plugin/plugin.py", line 304, in onCommand
2024-02-06 21:58:03.924 Error: Tiny Tuya: UpdateDevice(DeviceID, 3, Level, 1, 0)
2024-02-06 21:58:03.924 Error: Tiny Tuya: File "/home/pi/domoticz/plugins/Domoticz-TinyTUYA-Plugin/plugin.py", line 2987, in UpdateDevice
2024-02-06 21:58:03.924 Error: Tiny Tuya: Devices[ID].Units[Unit].LastLevel = sValue
```

**Comment by Xenomes** [comment]
Ha i see a problem😅 can you edit the thermostat device and change it to:
![afbeelding](https://github.com/Xenomes/Domoticz-TinyTUYA-Plugin/assets/11648297/dec3a827-22fe-447a-b73a-24c2df8ec80e)

**Comment by jaapkoppejan** [comment]
I changed it, however it does not resolve the problem of lack of sensor data.

**Comment by Xenomes** [comment]
Can you put the plugin on all debug. Then restart the plugin and look for update or error lines. Was this a new device or was it already in the systeem?

**Comment by Xenomes** [comment]
Can you check out the Develop? I made some modifications.
```
cd ~/domoticz/plugins/Domoticz-TinyTUYA-Plugin
git pull
git checkout Develop
sudo systemctl restart domoticz.service
```
To switch back to the Master version, change 'Develop' to 'Master'.

**Comment by jaapkoppejan** [comment]
Hi Xenomes,
There seem to be less errors now with this development version. The sole error left in the log is
`2024-02-07 22:18:49.857 Error: TinyTuya: Device read failed: bf56544b14bcff04e5irmc`
In practise however, the domoticz sensors of the device are still not updated. For the temperature sensor and the thermostat, values in the JSON file seem a factor of 10 too high (this might be related to the min/max settings in the thermostat) .  When I enter a value in the thermostat of the on/off swich in Domoticz, it does result in the same change in real life.
The device is new in my system, only with your plugin I am trying to get it into Domoticz. 

My log now says (username and password anonimised here);
```
024-02-07 22:31:36.167 TinyTuya: TinyTUYA 1.7.4 plugin started
2024-02-07 22:31:36.168 TinyTuya: TinyTuyaVersion:1.13.1
2024-02-07 22:31:36.168 TinyTuya: Debug logging mask set to: PYTHON PLUGIN QUEUE IMAGE DEVICE CONNECTION MESSAGE
2024-02-07 22:31:36.168 TinyTuya: 'HardwareID':'35'
2024-02-07 22:31:36.168 TinyTuya: 'HomeFolder':'/home/pi/domoticz/plugins/Domoticz-TinyTUYA-Plugin/'
2024-02-07 22:31:36.168 TinyTuya: 'StartupFolder':'/home/pi/domoticz/'
2024-02-07 22:31:36.168 TinyTuya: 'UserDataFolder':'/home/pi/domoticz/'
2024-02-07 22:31:36.168 TinyTuya: 'Database':'/home/pi/domoticz/domoticz.db'
2024-02-07 22:31:36.168 TinyTuya: 'Language':'nl'
2024-02-07 22:31:36.168 TinyTuya: 'Version':'1.7.4'
2024-02-07 22:31:36.168 TinyTuya: 'Author':'Xenomes'
2024-02-07 22:31:36.168 TinyTuya: 'Name':'TinyTuya'
2024-02-07 22:31:36.168 TinyTuya: 'Port':'0'
2024-02-07 22:31:36.168 TinyTuya: 'Username':'XXXXXXXX'
2024-02-07 22:31:36.168 TinyTuya: 'Password':'XXXXXXXXX'
2024-02-07 22:31:36.168 TinyTuya: 'Key':'tinytuya'
2024-02-07 22:31:36.168 TinyTuya: 'Mode1':'eu'
2024-02-07 22:31:36.168 TinyTuya: 'Mode2':'bf56544b14bcff04e5irmc'
2024-02-07 22:31:36.168 TinyTuya: 'Mode6':'-1'
2024-02-07 22:31:36.168 TinyTuya: 'DomoticzVersion':'2024.4'
2024-02-07 22:31:36.168 TinyTuya: 'DomoticzHash':'731719420'
2024-02-07 22:31:36.168 TinyTuya: 'DomoticzBuildTime':'2024-01-30 09:16:34'
2024-02-07 22:31:36.168 TinyTuya: Device count: 2
2024-02-07 22:31:36.168 TinyTuya: Device ID: '70116356840d8e5d4b96'
2024-02-07 22:31:36.168 TinyTuya: --->Unit Count: '1'
2024-02-07 22:31:36.168 TinyTuya: --->Unit: 1
2024-02-07 22:31:36.168 TinyTuya: --->Unit Name: 'Pi meterkast'
2024-02-07 22:31:36.168 TinyTuya: --->Unit nValue: 0
2024-02-07 22:31:36.168 TinyTuya: --->Unit sValue: 'Off'
2024-02-07 22:31:36.168 TinyTuya: --->Unit LastLevel: 0
2024-02-07 22:31:36.169 TinyTuya: Device ID: 'bf56544b14bcff04e5irmc'
2024-02-07 22:31:36.169 TinyTuya: --->Unit Count: '7'
2024-02-07 22:31:36.169 TinyTuya: --->Unit: 1
2024-02-07 22:31:36.169 TinyTuya: --->Unit Name: 'IR verwarming ketelhuis (Power)'
2024-02-07 22:31:36.169 TinyTuya: --->Unit nValue: 0
2024-02-07 22:31:36.169 TinyTuya: --->Unit sValue: 'Off'
2024-02-07 22:31:36.169 TinyTuya: --->Unit LastLevel: 0
2024-02-07 22:31:36.169 TinyTuya: --->Unit: 2
2024-02-07 22:31:36.169 TinyTuya: --->Unit Name: 'IR verwarming ketelhuis (Temperature)'
2024-02-07 22:31:36.169 TinyTuya: --->Unit nValue: 0
2024-02-07 22:31:36.169 TinyTuya: --->Unit sValue: ''
2024-02-07 22:31:36.169 TinyTuya: --->Unit LastLevel: 0
2024-02-07 22:31:36.169 TinyTuya: --->Unit: 3
2024-02-07 22:31:36.169 TinyTuya: --->Unit Name: 'IR verwarming ketelhuis (Thermostat)'
2024-02-07 22:31:36.169 TinyTuya: --->Unit nValue: 0
2024-02-07 22:31:36.169 TinyTuya: --->Unit sValue: ''
2024-02-07 22:31:36.169 TinyTuya: --->Unit LastLevel: 0
2024-02-07 22:31:36.169 TinyTuya: --->Unit: 4
2024-02-07 22:31:36.169 TinyTuya: --->Unit Name: 'IR verwarming ketelhuis (Mode)'
2024-02-07 22:31:36.169 TinyTuya: --->Unit nValue: 1
2024-02-07 22:31:36.169 TinyTuya: --->Unit sValue: '10'
2024-02-07 22:31:36.169 TinyTuya: --->Unit LastLevel: 10
2024-02-07 22:31:36.169 TinyTuya: --->Unit: 5
2024-02-07 22:31:36.169 TinyTuya: --->Unit Name: 'IR verwarming ketelhuis (Window check)'
2024-02-07 22:31:36.169 TinyTuya: --->Unit nValue: 1
2024-02-07 22:31:36.169 TinyTuya: --->Unit sValue: 'On'
2024-02-07 22:31:36.169 TinyTuya: --->Unit LastLevel: 0
2024-02-07 22:31:36.169 TinyTuya: --->Unit: 6
2024-02-07 22:31:36.169 TinyTuya: --->Unit Name: 'IR verwarming ketelhuis (Child lock)'
2024-02-07 22:31:36.169 TinyTuya: --->Unit nValue: 0
2024-02-07 22:31:36.169 TinyTuya: --->Unit sValue: 'Off'
2024-02-07 22:31:36.169 TinyTuya: --->Unit LastLevel: 0
2024-02-07 22:31:36.169 TinyTuya: --->Unit: 7
2024-02-07 22:31:36.169 TinyTuya: --->Unit Name: 'IR verwarming ketelhuis (Eco)'
2024-02-07 22:31:36.169 TinyTuya: --->Unit nValue: 0
2024-02-07 22:31:36.169 TinyTuya: --->Unit sValue: ''
2024-02-07 22:31:36.169 TinyTuya: --->Unit LastLevel: 0
2024-02-07 22:31:36.169 TinyTuya: Pushing 'PollIntervalDirective' on to queue
```

**Comment by Xenomes** [comment]
Can you delete the create devices and reload the plugin?

**Comment by jaapkoppejan** [comment]
Yes, I just did. However with the same result; devices appear again but with no values
![image](https://github.com/Xenomes/Domoticz-TinyTUYA-Plugin/assets/40127840/ee1daa6a-d0f2-4aab-913d-ecea1bd15a78)

I will send you a private message with access data to the devices in my Tuya account so you can play with it yourself. Both devices in my Tuya account are now disconnected from a load, so you can't mess up anything in my house :)

**Comment by Xenomes** [comment]
You can send it to xenomes@outlook.com in dutch 😉

**Comment by jaapkoppejan** [comment]
Thanks for your support, it now works on a raspberry pi 3b

**Comment by Xenomes** [comment]
I will go look further to find out why the bookworm behaves like that.

**Comment by Xenomes** [comment]
Fixed in 1.7.5
